qiita_pet/test/test_qiita_redbiom.py

```diff
@@ -0,0 +1,161 @@
+        post_args = {
+            'search': 'Diesel',
+            'search_on': 'metadata'
+        }
+        response = self.post('/redbiom/', post_args)
+        self.assertEqual(response.code, 200)
+        exp = {'status': 'success', 'message': '', 'data': OBSERVATION}
+        self.assertEqual(loads(response.body), exp)
+
+        post_args = {
+            'search': 'inf',
+            'search_on': 'metadata'
+        }
+        response = self.post('/redbiom/', post_args)
+        self.assertEqual(response.code, 200)
+        exp = {'status': 'success',
+               'message': 'No samples where found! Try again ...', 'data': []}
+        self.assertEqual(loads(response.body), exp)
+
+        post_args = {
+            'search': '4353076',
+            'search_on': 'metadata'
+        }
+        response = self.post('/redbiom/', post_args)
+        self.assertEqual(response.code, 200)
+        exp = {'status': 'success',
+               'message': ('Not a valid search: "4353076", are you sure this '
+                           'is a valid metadata value?'), 'data': []}
+        self.assertEqual(loads(response.body), exp)
+
+    def test_post_observations(self):
+        post_args = {
+            'search': ('4479944'),
+            'search_on': 'observations'
+        }
+        response = self.post('/redbiom/', post_args)
+        exp = {'status': 'success', 'message': '', 'data': SEQUENCE}
+        self.assertEqual(response.code, 200)
+        self.assertEqual(loads(response.body), exp)
+
+        post_args = {
+            'search': ('TT'),
+            'search_on': 'observations'
+        }
+        response = self.post('/redbiom/', post_args)
+        exp = {'status': 'success',
+               'message': 'No samples where found! Try again ...', 'data': []}
+        self.assertEqual(response.code, 200)
+        self.assertEqual(loads(response.body), exp)
+
+    def test_post_categories(self):
+        post_args = {
+            'search': ('texture'),
+            'search_on': 'categories'
+        }
+        response = self.post('/redbiom/', post_args)
+        exp = {'status': 'success', 'message': '', 'data': CATEGORIES}
+        self.assertEqual(response.code, 200)
+        self.assertEqual(loads(response.body), exp)
+
+        post_args = {
+            'search': ('longtext'),
+            'search_on': 'categories'
+        }
+        response = self.post('/redbiom/', post_args)
+        exp = {'status': 'success',
+               'message': 'No samples where found! Try again ...', 'data': []}
+        self.assertEqual(response.code, 200)
+        self.assertEqual(loads(response.body), exp)
+
+        post_args = {
+            'search': '4353076',
+            'search_on': 'categories'
+        }
+        response = self.post('/redbiom/', post_args)
+        self.assertEqual(response.code, 200)
+        exp = {'status': 'success',
+               'message': ('Not a valid search: "4353076", are you sure this '
```

Comment:
In order to allow this to move forward, can this be added as a TODO item (even if it is only a discussion)? I don't understand why a number will not be a valid stem, or why searching over numerical values is not allowed. However, I don't want to block this PR for this discussion.